index.js

```diff
@@ -55,6 +57,7 @@ function ensureAllCommentsPrinted(astComments) {
 }
 
 function formatWithCursor(text, opts, addAlignmentSize) {
+  text = prependFormatIfNecessary(text);
```

Comment:
> There's also the cursorOffset option -- it's worth ensuring that it will take the added docblock into account when computing the new offset after formatting

the rest i'm on board with and plan on doing.  this i feel less passionate about